src/components/molecules/FormFields/CustomProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactElement, useState, useEffect } from 'react'
+import { useField } from 'formik'
+import { toast } from 'react-toastify'
+import CustomInput from './URLInput/Input'
+import { useOcean } from '../../../providers/Ocean'
+import { InputProps } from '../../atoms/Input'
+import { Ocean } from '@oceanprotocol/lib'
+
+export default function CustomProvider(props: InputProps): ReactElement {
+  const [field, meta, helpers] = useField(props.name)
+  const [isLoading, setIsLoading] = useState(false)
+  const [providerUrl, setProviderUrl] = useState<string>()
+  const { config } = useOcean()
+
+  function loadProvider() {
+    if (!providerUrl) return
+    async function validateProvider() {
+      let valid: boolean
+      try {
+        setIsLoading(true)
+        const ocean: Ocean = await Ocean.getInstance(config)
```

Comment:
`ocean` is always there:

```js
const { ocean } = useOcean()
```

Reply:
Thanks for pointing that out. I've changed it to `const { ocean } = useOcean()`